backend/src/test/java/de/armbrust/planz/service/ProductServiceTest.java

```diff
@@ -124,12 +126,105 @@ public void testAddingListOfNotExistingProducts() throws CryptoException, Missin
+    public void IfProductAvailibeInventoryShouldAdded() {
+        //GIVEN
+        Product product1 = Product.builder()
+                .sku("123")
+                .title("Product1")
+                .asin("ABC")
+                .inventory(new ArrayList<>())
+                .build();
+
+        Inventory inventory1 = Inventory.builder()
+                .warehouse("AnyWarehouse")
+                .condition("NEW")
+                .country("DE")
+                .amount("23")
+                .date("2021-02-25")
+                .sku("123")
+                .build();
+
+        when(productMongoDb.findById(inventory1.getSku())).thenReturn(Optional.of(product1));
+
+        // WHEN
+        productService.findProductAndAddInventory(inventory1);
+
+        // THEN
+        verify(productMongoDb).findById("123");
+        verify(productMongoDb).save(product1);
+        assertThat(product1.getInventory(), containsInAnyOrder(inventory1));
+    }
+
+
+    @Test
+    @DisplayName("if product is not available nothing should happen")
+    public void IfProductNotAvailibeNothingHappens() {
+        //GIVEN
+        Product product1 = Product.builder()
+                .sku("123")
+                .title("Product1")
+                .asin("ABC")
+                .inventory(new ArrayList<>())
+                .build();
+
+        Inventory inventory1 = Inventory.builder()
+                .warehouse("AnyWarehouse")
+                .condition("NEW")
+                .country("DE")
+                .amount("23")
+                .date("2021-02-25")
+                .sku("123")
+                .build();
+
+        when(productMongoDb.findById(inventory1.getSku())).thenReturn(Optional.empty());
+
+        // WHEN
+        productService.findProductAndAddInventory(inventory1);
+
+        // THEN
+        verify(productMongoDb, never()).save(any());
+    }
+
+    @Test
+    @DisplayName("if inventory DataPoint is already available, it not be added")
+    public void IfProductHasInventoryPointAlreadyItShouldNotBeAdded() {
+        //GIVEN
+        Inventory inventory1 = Inventory.builder()
+                .warehouse("AnyWarehouse")
+                .condition("NEW")
+                .country("DE")
+                .amount("23")
+                .date("2021-02-25")
+                .sku("123")
+                .build();
+
+        ArrayList<Inventory> inventoryArray = new ArrayList<>();
+        inventoryArray.add(inventory1);
+
+        Product product1 = Product.builder()
+                .sku("123")
+                .title("Product1")
+                .asin("ABC")
+                .inventory(inventoryArray)
+                .build();
+
+        when(productMongoDb.findById(inventory1.getSku())).thenReturn(Optional.of(product1));
+
+        // WHEN
+        productService.findProductAndAddInventory(inventory1);
+
+        // THEN
+        assertThat(product1.getInventory().toString().length(), is(inventoryArray.toString().length()));
```

Comment:
this should not work as updates are only made on the newly created `updatedProduct` and saved to your hypothetical DB 
see comment in your ProductService for better testability which would allow you to use `verify()` here as in your test above